src/webui/components/Search/index.js

```diff
@@ -0,0 +1,190 @@
+/**
+ * @prettier
+ * @flow
+ */
+
+import React, { Component } from 'react';
+import type { Node } from 'react';
+
+import { default as IconSearch } from '@material-ui/icons/Search';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import debounce from 'lodash/debounce';
+
+import API from '../../utils/api';
+import AutoComplete from '../AutoComplete';
+import colors from '../../utils/styles/colors';
+import { getDetailPageURL } from '../../utils/url';
+
+import { IProps, IState } from './types';
+
+const CONSTANTS = {
+  API_DELAY: 300,
+  PLACEHOLDER_TEXT: 'Search Packages',
+  ABORT_ERROR: 'AbortError',
+};
+
+class Search extends Component<IProps, IState> {
+  requestList: Array<any>;
+  cancelAllSearchRequests: Function;
+  handlePackagesClearRequested: Function;
+  handleSearch: Function;
+  handleClickSearch: Function;
```

Comment:
Please, could we have an interface for the handlers?

Reply:
Done :)